todo-app/src/App.js

```diff
@@ -1,20 +1,135 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import todo from './todo.json';
+import TasksForm  from './components/TasksForm';
+import AddTodo from './components/AddTodo';
+import DeleteAllButton from './components/DeleteAllButton';
+
 
 class App extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      todos: [
+        {
+          description: "Study Math",
+          done: false,
+          id:1
+        },
+        {
+          description: "Finish your homework",
+          done: true,
+          id:2
+        },
+        {
+          description: "Eat your sandwich",
+          done: true,
+          id:3
+        },
+        {
+          description: "Listen to music",
+          done: true,
+          id:4
+        }
+      ],
+      lastGeneratedId:4
+    }
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleDelete = this.handleDelete.bind(this)
+    this.handleCompletedTask = this.handleCompletedTask.bind(this)
+    this.handleNewTaskSubmit = this.handleNewTaskSubmit.bind(this)
+    this.handleDeleteAll = this.handleDeleteAll.bind(this)
+  }
+
+handleSubmit(newDescription,id){
+  const index = this.state.todos.findIndex((item) => {
+    return item.id === id
+  })
+  const todo = this.state.todos[index]
+  const newTodo = Object.assign({}, todo, {
+    description: newDescription
+  })
+  this.setState({
+    todos: this.state.todos.slice(0, index).concat(newTodo, this.state.todos.slice(index + 1))
+  })
+  console.log(index)
+  console.log(newDescription + id)
+}
+
+handleDelete(newId){
+  console.log(newId)
+  const index = this.state.todos.findIndex((item) => {
+    return item.id === newId}
+  )
+  this.setState({
+    todos: this.state.todos.slice(0, index).concat(this.state.todos.slice(index + 1))
+  })
+  console.log(index)
+  console.log(this.state.todos)
+
+}
+
+handleCompletedTask(newId){
+  console.log(newId)
+  const index = this.state.todos.findIndex((item) => {
+    return item.id === newId}
+  )
+  const completed = this.state.todos[index].done ? false : true;
```

Comment:
Although this is eventually consistent and it works, you are not using the value directly that is coming from the onChange event of the checkbox, but you are reading the one from the state. You might get yourself in trouble doing that. How would you use the value that is coming from the onChange event from the checkbox?